tests/php/Control/CookieTest.php

```diff
@@ -201,4 +206,57 @@ public function testForceExpiry()
         $this->assertEmpty(Cookie::get('newCookie'));
         $this->assertEmpty(Cookie::get('newCookie', false));
     }
+
+    /**
+     * Check whether warnings are correctly logged for insecure requests
+     */
+    public function testValidateSameSite(): void
+    {
+        // Throw an exception when a warning is logged so we can catch it
+        $mockLogger = $this->getMockBuilder(Logger::class)->setConstructorArgs(['testLogger'])->getMock();
+        $catchMessage = 'A warning was logged';
+        $mockLogger->expects($this->once())
+            ->method('warning')
+            ->willThrowException(new Exception($catchMessage));
+        Injector::inst()->registerService($mockLogger, LoggerInterface::class);
+
+        // Only samesite === 'None' should log a warning on insecure requests
+        Director::config()->set('alternate_base_url', 'http://insecure.example.com/');
+        $warningLogged = false;
+        try {
+            Cookie::validateSameSite('None');
+        } catch (Exception $e) {
+            if ($e->getMessage() !== $catchMessage) {
+                throw $e;
+            }
+            $warningLogged = true;
+        }
+        $this->assertTrue($warningLogged);
+        Cookie::validateSameSite('Lax');
+        Cookie::validateSameSite('Strict');
+
+        // There should be no warnings logged for secure requests
+        Director::config()->set('alternate_base_url', 'https://secure.example.com/');
+        Cookie::validateSameSite('None');
```

Comment:
This is pretty weird.  I'm thinking update Cookie::validateSameSite() to return a bool so it's more testable and behaves like people normally expect a valiate*() function to operate, even though we don't actually use the validate return value

Reply:
I don't think we should return a value just because other similarly named methods return values.... and it wouldn't make it more testable so much as it would lend itself to writing incorrect tests (testing a return value that is never used is a great way to have tests passing when functionality is broken).
Maybe it should be renamed instead? Though I can't think of another name that makes more sense.